kotlin/display-feature-layers/src/main/java/com/esri/arcgisruntime/sample/displayfeaturelayers/MainActivity.kt

```diff
@@ -0,0 +1,263 @@
+        geodatabase.loadAsync()
+        geodatabase.addDoneLoadingListener {
+            if (geodatabase.loadStatus == LoadStatus.LOADED) {
+                // get the feature table with the name
+                val geodatabaseFeatureTable = geodatabase.getGeodatabaseFeatureTable("Trailheads")
+                // create a feature layer with the feature table
+                val featureLayer = FeatureLayer(geodatabaseFeatureTable)
+                // set the viewpoint to Malibu, California
+                val viewpoint = Viewpoint(34.0772, -118.7989, 600000.0)
+                // set the feature layer on the map
+                setFeatureLayer(featureLayer, viewpoint)
+            } else {
+                showError(geodatabase.loadError.message)
+            }
+        }
+    }
+
+    /**
+     * Load a feature layer with a local geopackage file
+     */
+    private fun loadGeopackage() {
+        // locate the .gpkg file in the device
+        val geopackageFile = File(getExternalFilesDir(null), "/AuroraCO.gpkg")
+        // instantiate the geopackage with the file path
+        val geoPackage = GeoPackage(geopackageFile.path)
+        // load the geopackage
+        geoPackage.loadAsync()
+        geoPackage.addDoneLoadingListener {
+            if (geoPackage.loadStatus == LoadStatus.LOADED) {
+                // get the first feature table in the geopackage
+                val geoPackageFeatureTable = geoPackage.geoPackageFeatureTables.first()
+                // create a feature layer with the feature table
+                val featureLayer = FeatureLayer(geoPackageFeatureTable)
+                // set the viewpoint to Denver, CO
+                val viewpoint = Viewpoint(39.7294, -104.8319, 500000.0)
+                // set the feature layer on the map
+                setFeatureLayer(featureLayer, viewpoint)
+            } else {
+                showError(geoPackage.loadError.message)
+            }
+        }
+    }
+
+    /**
+     * Load a feature layer with a local shapefile file
+     */
+    private fun loadShapefile() {
+        try {
+            // locate the shape file in device
+            val file = File(
+                getExternalFilesDir(null),
+                "/ScottishWildlifeTrust_ReserveBoundaries_20201102.shp"
+            )
+            // create a shapefile feature table from a named bundle resource
+            val shapeFileTable = ShapefileFeatureTable(file.path)
+            // create a feature layer for the shapefile feature table
+            val featureLayer = FeatureLayer(shapeFileTable)
+            // set the viewpoint to Scotland
+            val viewpoint = Viewpoint(56.641344, -3.889066, 6000000.0)
+            // set the feature layer on the map
+            setFeatureLayer(featureLayer, viewpoint)
+        } catch (e: Exception) {
+            showError(e.message)
+        }
+    }
+
+    /**
+     * Sets the map using the loaded [layer] at the given [viewpoint]
```

Comment:
Hmm, maybe I'm mistaken but I think adding a layer to map which is set on a MapView will cause the layer to begin loading. So even if you call this method with an unloaded layer it's fine because the layer will load when it's added to the map. For example in `loadFeatureServiceURL` I don't see any explicit loading happening but the layer still loads and displays when it's selected in the app.